frontend/src/utils/useDashboardSidebar.ts

```diff
@@ -0,0 +1,328 @@
+    const { name: current } = route;
+    const isActiveRoute =
+      item.name === current?.toString() ||
+      current?.toString().startsWith(`${item.name}.`);
+    const classes: string[] = [];
+    if (isActiveRoute) {
+      classes.push("router-link-active", "bg-link-hover");
+      return classes;
+    }
+    if (
+      item.name === WORKSPACE_ROUTE_USERS &&
+      current?.toString() === WORKSPACE_ROUTE_USER_PROFILE
+    ) {
+      classes.push("router-link-active", "bg-link-hover");
+      return classes;
+    }
+    return classes;
+  };
+
+  const filterSidebarByPermissions = (
+    sidebarList: DashboardSidebarItem[]
+  ): DashboardSidebarItem[] => {
+    return sidebarList
+      .filter((item) => {
+        const routeConfig = flattenRoutes.value.find(
+          (workspaceRoute) => workspaceRoute.name === item.name
+        );
+        return (routeConfig?.permissions ?? []).every((permission) =>
+          hasWorkspacePermissionV2(permission)
+        );
+      })
+      .map((item) => ({
+        ...item,
+        expand:
+          item.expand ||
+          (item.children ?? [])
+            .reduce((classList, child) => {
+              classList.push(...getItemClass(child));
+              return classList;
+            }, [] as string[])
+            .includes("router-link-active"),
+        children: filterSidebarByPermissions(item.children ?? []),
+      }));
+  };
+
+  const getFlattenRoutes = (
+    routes: RouteRecordRaw[],
+    permissions: Permission[] = []
+  ): {
+    name: string;
+    permissions: Permission[];
+  }[] => {
+    return routes.reduce(
+      (list, workspaceRoute) => {
+        const requiredWorkspacePermissionListFunc =
+          workspaceRoute.meta?.requiredWorkspacePermissionList;
+        let requiredPermissionList = requiredWorkspacePermissionListFunc
+          ? requiredWorkspacePermissionListFunc()
+          : [];
+        if (requiredPermissionList.length === 0) {
+          requiredPermissionList = permissions;
+        }
+
+        if (workspaceRoute.name && workspaceRoute.name.toString() !== "") {
+          list.push({
+            name: workspaceRoute.name.toString(),
+            permissions: requiredPermissionList,
+          });
+        }
+        if (workspaceRoute.children) {
+          list.push(
+            ...getFlattenRoutes(workspaceRoute.children, requiredPermissionList)
+          );
+        }
+        return list;
+      },
+      [] as { name: string; permissions: Permission[] }[]
+    );
+  };
+
+  const flattenRoutes = computed(() => {
+    return getFlattenRoutes(workspaceRoutes);
+  });
+
+  const dashboardSidebarItemList = computed((): DashboardSidebarItem[] => {
+    const sidebarList: DashboardSidebarItem[] = [
+      {
+        navigationId: "bb.navigation.home",
+        title: t("common.home"),
+        icon: () => h(HomeIcon),
+        name: WORKSPACE_ROUTE_LANDING,
+        type: "route",
+        shortcuts: ["g", "h"],
+      },
```

Comment:
We can even remove the "Home" in the sidebar, for example:

![CleanShot 2024-11-26 at 22 20 24@2x](https://github.com/user-attachments/assets/7bf6b335-9836-469f-9028-61ab41c41969)

Reply:
We can do it in a latter change. Let's get a feeling about other changes first.